### DISCUSSION TITLE
TypeError: Converting circular structure to JSON on integration with Alpine.js

### DISCUSSION THREAD

**Discussion by feryardiant** [livewire/livewire#5285] [1. Help]
First of all, I got the following error in the console

```
Alpine Expression Error: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    --- property 'defer' closes the circle

Expression: "show"
```

and here's my component's structure looks like these

```php
// app/Http/Livewire/SomeComponent.php
class SomeComponent extends Livewire\Component
{
    public $showModalSearch = false;

    public function render()
    {
        return view('livewire.some-component');
    }
}
```

```blade
// resources/views/livewire/some-component.blade.php
<x-modal wire:model="showModalSearch">
    <livewire:reports.modal.search />
</x-modal>
// ... other codes
```

```blade
// resources/components/modal.blade.php
<div x-data="{ show: @entangle($attributes->wire('model')) }">
    {{-- other codes --}}
</div>
```
As [documentation](https://laravel-livewire.com/docs/2.x/alpine-js#livewire-directives-from-blade-components) suggested `@entabgle(...)`, I expect it would return the value of `showModalSearch` model.

Then I look at the HTML output, my `modal.blade.php` become something like this

```html
<div x-data="{ show: window.Livewire.find('xbC2uudJBER4L5QQ0kFz').entangle('showModalSearch') }">
   <!-- other codes -->
</div>
```

and here's the output of `Livewire.find('...').entangle('...')` above in console
![Screen Shot 2022-10-04 at 05 40 09](https://user-images.githubusercontent.com/508665/193698499-4d0865c2-df19-41cb-88e3-4a46dc015b62.png)

Is that the actual behavior of `@entangle()`?

Any helps appreciated

**Comment by joshhanley** [comment]
@feryardiant that's correct, entangle is just a reference object so you can't actually see the value of the entangled object. Under the hood, when Alpine is initialised,  looks at the `livewireEntangle` property on the object and then talks directly to Livewire to access the property through the value of `livewireEntangle`.

Hope this helps!
